v21.1/cockroach-userfile-upload.md

```diff
@@ -125,6 +125,126 @@ cockroach userfile upload /Users/maxroach/Desktop/test-data.csv userfile://testd
+downloaded database-backup/BACKUP-CHECKPOINT-CHECKSUM to database-backup/BACKUP-CHECKPOINT-CHECKSUM (4 B)
+downloaded database-backup/BACKUP-STATISTICS to database-backup/BACKUP-STATISTICS (53 KiB)
+downloaded database-backup/BACKUP_MANIFEST to database-backup/BACKUP_MANIFEST (2.6 KiB)
+downloaded database-backup/BACKUP_MANIFEST-CHECKSUM to database-backup/BACKUP_MANIFEST-CHECKSUM (4 B)
+downloaded database-backup/data/679645557047099393.sst to database-backup/data/679645557047099393.sst (5.8 KiB)
+. . .
+downloaded database-backup/data/679645558404448257.sst to database-backup/data/679645558404448257.sst (5.8 KiB)
+downloaded database-backup/data/679645558408249345.sst to database-backup/data/679645558408249345.sst (41 KiB)
+~~~
+
+At this point, you have backed up the database to `userfile` storage and you have a copy of that backup on your local machine.
+
+Your backup will contain files similar to the following structure:
+
+~~~
+database-backup/
+├── BACKUP-CHECKPOINT-679645548255936513-CHECKSUM
+├── BACKUP-CHECKPOINT-CHECKSUM
+├── BACKUP-STATISTICS
+├── BACKUP_MANIFEST
+├── BACKUP_MANIFEST-CHECKSUM
+└── data
+    ├── 679645557047099393.sst
+    ├── 679645557048475649.sst
+    ├── 679645557048868865.sst
+    ├── 679645558151741442.sst
+. . .
+~~~
+
+If you should need to recover that data, or, apply it for testing purposes, you can now upload these files to a cluster with `cockroach userfile upload` and [`RESTORE`](restore.html).
+
+For the purposes of this example, [`DROP`](drop-database.html) the database:
+
+~~~sql
+DROP DATABASE movr CASCADE;
+~~~
+
+And then, use [`cockroach userfile delete`](cockroach-userfile-delete.html) to remove this from `userfile`:
+
+~~~shell
+cockroach userfile delete 'userfile://defaultdb.public.userfiles_$user/database-backup' --url 'postgresql://root@localhost:26257?sslmode=disable'
+~~~
+
+~~~
+successfully deleted database-backup/BACKUP-CHECKPOINT-679645548255936513-CHECKSUM
+successfully deleted database-backup/BACKUP-CHECKPOINT-CHECKSUM
+successfully deleted database-backup/BACKUP-STATISTICS
+successfully deleted database-backup/BACKUP_MANIFEST
+successfully deleted database-backup/BACKUP_MANIFEST-CHECKSUM
+successfully deleted database-backup/data/679645557047099393.sst
+. . .
+successfully deleted database-backup/data/679645558154264579.sst
+~~~
+
+Currently, `cockroach userfile upload` will not recursively upload files from a directory. See this [known limitation](../{{site.versions["stable"]}}/cockroach-userfile-upload.html#known-limitation). It is possible to programmatically upload your files from the command line.
+
+For example, the following command finds all files under `database-backup` and pipes them through to [`xargs`](https://linux.die.net/man/1/xargs), which will execute `cockroach userfile upload` for each file. Every occurrence of `{}` in the command, is replaced with the filename.
```

Comment:
(ok I just looked this up and it's the latter: https://unix.stackexchange.com/questions/616899/xargs-replstr-not-working-as-expected-not-accepted-as-a-second-argument)

in that case I recommend removing this sentence and letting users read the `xargs` man page / etc.  on the reasoning that by saying just a little about it is possibly more confusing than just telling the user "use this exact command" and leaving it at that.  by mentioning it, I think there is something there I need to understand (but in this case AFAICT I don't)

Reply:
further comments:

1. suggest editing to even higher level e.g. "the following command finds all files under ... and runs `cockroach userfile upload` on each file ..."  and folks can read the script / read about `xargs` elsewhere if they care.  

2. Alternatively if we are mentioning `xargs` and linking to it we should do the same for `find`.  However I would maybe recommend the approach 1 above since there are many different `find` s out there, and probably many `xargs` as well